scripts/sap_automation_qa.sh

```diff
@@ -100,24 +147,137 @@ get_playbook_name() {
+    set -e  # Re-enable exit on error
+
+    if [[ $az_exit_code -ne 0 ]]; then
+        extracted_message=$(extract_error_message "$error_message")
+        log "ERROR" "Azure CLI error: $extracted_message"
+        exit 1
+    fi
+    
+    # Attempt to retrieve the secret value and handle errors
+    log "INFO" "Retrieving secret '$secret_name' from Key Vault '$key_vault_name'..."
+    set +e  # Temporarily disable exit on error
+    secret_value=$(az keyvault secret show --vault-name "$key_vault_name" --name "$secret_name" --query "value" -o tsv 2>&1)
+    az_exit_code=$?  # Capture the exit code of the az command
+    set -e  # Re-enable exit on error
+
+    if [[ $az_exit_code -ne 0 ]]; then
+        extracted_message=$(extract_error_message "$secret_value")
+        log "ERROR" "Failed to retrieve secret '$secret_name' from Key Vault '$key_vault_name': $extracted_message"
+        exit 1
+    fi
+
+    log "INFO" "Successfully retrieved secret from Key Vault."
+    temp_file=$(mktemp --suffix=.ppk)
+    echo "$secret_value" > "$temp_file"
+    log "INFO" "Temporary SSH key file created: $temp_file"
+}
+
+"""
+Run the ansible playbook.
+
+:param playbook_name: The name of the playbook to run.
+:param system_hosts: The path to the inventory file.
+:param system_params: The path to the SAP parameters file.
+:param auth_type: The authentication type (e.g., "SSHKEY", "VMPASSWORD").
+:param system_config_folder: The path to the system configuration folder.
+:param secret_name: The name of the secret in the Key Vault.
+:return: None. Exits with the return code of the ansible-playbook command.
+"""
 run_ansible_playbook() {
     local playbook_name=$1
     local system_hosts=$2
     local system_params=$3
     local auth_type=$4
     local system_config_folder=$5
+    local secret_name=$6
 
     if [[ "$auth_type" == "SSHKEY" ]]; then
-        local ssh_key="${cmd_dir}/../WORKSPACES/SYSTEM/$SYSTEM_CONFIG_NAME/ssh_key.ppk"
-        log "INFO" "Using SSH key: $ssh_key."
-        command="ansible-playbook ${cmd_dir}/../src/$playbook_name.yml -i $system_hosts --private-key $ssh_key \
-        -e @$VARS_FILE -e @$system_params -e '_workspace_directory=$system_config_folder'"
+        log "INFO" "Authentication type is SSHKEY."
+
+        # Extract key_vault_id from sap-parameters.yaml
+        key_vault_id=$(grep "^key_vault_id:" "$system_params" | awk '{split($0,a,": "); print a[2]}' | xargs)
+        if [[ -z "$key_vault_id" ]]; then
+            log "ERROR" "Error: key_vault_id is not defined in $system_params."
+            exit 1
+        fi
+        log "INFO" "Extracted key_vault_id: $key_vault_id"
+
+        # Extract Key Vault details and check MSI permissions
+        check_msi_permissions "$key_vault_id"
+        if [[ -n "$key_vault_name" && -n "$secret_name" ]]; then
+            log "INFO" "Using Key Vault for SSH key retrieval."
+            log "INFO" "Temporary SSH key file: $temp_file"
```

Comment:
will be good to check if the file exists or not, there's an existing function you can leverage.